src/rust/lib/system/web/src/lib.rs

```diff
@@ -503,7 +503,35 @@ pub async fn sleep(duration:Duration) {
 #[cfg(not(target_arch = "wasm32"))]
 pub use async_std::task::sleep;
 
+/// Stores arguments extracted from Location search. e.g. http://localhost/?arg0=value0&arg1=value1.
+#[derive(Shrinkwrap,Debug)]
+pub struct Arguments {
+    pub hash_map : HashMap<String,String>
+}
+
+impl Arguments {
+    fn args_from_search(search:&str) -> HashMap<String,String> {
+        if search.chars().nth(0) == Some('?') {
+            search[1..].split("&").filter_map(|arg| {
+                let x : Vec<&str> = arg.split("=").collect();
+                match x.len() {
+                    1 => Some((x[0].to_string(), "".to_string())),
+                    2 => Some((x[0].to_string(), x[1].to_string())),
+                    _ => None
+                }
+            }).collect()
+        } else {
+            default()
+        }
+    }
 
+    /// Creates a new arguments map from location search.
+    pub fn new() -> Self {
+        let search   = window().location().search().unwrap_or_else(|_| String::from(""));
```

Comment:
please read docs to `Option`. There are many places in the code where you can use better methods than these. For example here, `unwrap_or_default()`.